client/rest/dialer.go

```diff
@@ -0,0 +1,118 @@
+package rest
```

Comment:
This code is unchanged, simply lifted and moved to a more appropriate place from `utils.go`